tests/test_commands.py

```diff
@@ -0,0 +1,249 @@
+serve_command = parametrized_fixture(serve=['serve'],
+                                     freeze_serve=['freeze', '--serve'])
+port = parametrized_fixture(8001, 8080)
+domain = parametrized_fixture('foo.bar', 'spam.eggs')
+protocol = parametrized_fixture('http', 'https')
+
+
+class ElsaRunner:
+    '''
+    Class for elsa fixture enabling blocking or background runs
+
+    If there is a local website.py in pwd, uses that one,
+    uses the one from fixtures instead.
+    '''
+    def run(self, *command):
+        print('COMMAND: python website.py', *command)
+        subprocess.run(self.create_command(command), check=True)
+
+    @contextmanager
+    def run_bg(self, *command):
+        print('COMMAND IN BACKGROUND: python website.py', *command)
+        proc = subprocess.Popen(self.create_command(command),
+                                stderr=subprocess.PIPE,
+                                universal_newlines=True)
+
+        # Wait for the server to start,
+        # i.e. wait for the first line on stderr:
+        #  * Running on http://127.0.0.1:8003/ (Press CTRL+C to quit)
+        sys.stderr.write(proc.stderr.readline())
+        # With the serve command, Flask is running in debug and restarts
+        # the server, so we'll also wait for next line:
+        #  * Restarting with stat
+        if command[0] == 'serve':
+            sys.stderr.write(proc.stderr.readline())
+
+        yield proc
+
+        os.kill(proc.pid, signal.SIGINT)  # Ctrl+C
+        try:
+            _, errs = proc.communicate(timeout=5)
+        except subprocess.TimeoutExpired:
+            proc.kill()
+            _, errs = proc.communicate()
+        sys.stderr.write(errs)
+        assert proc.returncode == 0
+
+    def finalize(self):
+        self.lax_rmtree(BUILDDIR_FIXTURES)
+
+    @classmethod
+    def create_command(cls, command):
+        if os.path.exists(SCRIPT):
+            script = SCRIPT
+        else:
+            script = SCRIPT_FIXTURES
+        command = tuple(str(item) for item in command)
+        return (sys.executable, script) + command
+
+    @classmethod
+    def lax_rmtree(cls, path):
+        try:
+            shutil.rmtree(path)
+        except FileNotFoundError:
+            pass
+
+
+@pytest.fixture
+def elsa():
+    er = ElsaRunner()
+    try:
+        yield er
+    finally:
+        er.finalize()
+
+
+def commit_info():
+    commit = str(sh.git('--no-pager', 'show', 'gh-pages', '--no-color'))
+    print(commit)
+    return commit
+
+
+def assert_commit_author(commit):
+    assert 'Author: Tester Elsa <tester@elsa.py>' in commit
+
+
+@pytest.fixture
+def gitrepo(tmpdir):
+    '''
+    This fixture creates a git repository with website.py in tmpdir
+    '''
+    repo = tmpdir.mkdir('repo')
+    script = repo.join(SCRIPT)
+    with open(SCRIPT_FIXTURES) as f:
+        script.write(f.read())
+    with repo.as_cwd():
+        sh.git.init()
+        sh.git.add(SCRIPT)
+        sh.git.config('user.email', 'tester@elsa.py')
```

Comment:
It's good practice to use a [reserved domain](https://tools.ietf.org/html/rfc2606) such as `example.com` or `elsa.invalid`, rather than something that might be a real address in the future.